common/src/main/java/org/apache/rocketmq/common/MixAll.java

```diff
@@ -122,8 +125,10 @@ public static String getDLQTopic(final String consumerGroup) {
 
     public static String brokerVIPChannel(final boolean isChange, final String brokerAddr) {
         if (isChange) {
-            String[] ipAndPort = brokerAddr.split(":");
-            String brokerAddrNew = ipAndPort[0] + ":" + (Integer.parseInt(ipAndPort[1]) - 2);
+            int split = brokerAddr.lastIndexOf(":");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the lastIndexOf() search on line 128 does not find the substring and returns -1, the call to substring() on line 129 will throw IndexOutOfBoundsException.